python/main.py

```diff
@@ -90,16 +107,69 @@ async def get_image(image_name):
 
     if not image.exists():
         logger.debug(f"Image not found: {image}")
-        image = images / "default.jpg"
+        image = images / "no_image.jpg"
 
     return FileResponse(image)
 
 
 class Item(BaseModel):
     name: str
     category: str
+    image: str 
+
+# hash the image file
+async def hash_and_save_image(image: UploadFile):
+    if not image.filename.endswith(".jpg"):
+        raise HTTPException(status_code=400, detail="Image path does not end with .jpg")
 
+    # create an instance of a hash function
+    sha256 = hashlib.sha256()
+
+    # get the binary data of the uploaded image
+    contents = await image.read()
+
+    # perform a hash calculation
+    sha256.update(contents)
+
+    # rename the file to a hash value
+    image_name = f"{sha256.hexdigest()}.jpg"
+
+    # save
+    image_path = images / image_name
+    with open(image_path, "wb") as f:
+        f.write(contents)
+
+    return image_name
 
 def insert_item(item: Item):
-    # STEP 4-1: add an implementation to store an item
-    pass
+        if not items_file.exists():
+            with open(items_file, "w") as file:
+                json.dump({"items": []}, file, indent=4) 
+        
+        # load the items.json file
+        with open(items_file, "r") as file:
+                data = json.load(file)
+
+        # add a new item to the "items" array
+        data["items"].append({"name": item.name, "category": item.category,"image_name": item.image_name})
+    
+        # save the updated data to items.json
+        with open(items_file, "w") as file:
+            json.dump(data, file, indent=4)
```

Comment:
Can we return a `200` response here with a message, something like `item added successfully`